models/code_samples.go

```diff
@@ -0,0 +1,127 @@
+package models
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// CodeSamples is a view model for the code samples markdown template. Using
+// these struct methods simplifies the logic needed in the markdown templates
+type CodeSamples struct {
+	Method *Method
+}
+
+func NewCodeSamples(method *Method) *CodeSamples {
+	return &CodeSamples{
+		Method: method,
+	}
+}
+
+func (c *CodeSamples) AppName() string {
+	return c.Method.Service.Pkg.App.Name
+}
+
+func (c *CodeSamples) PackageName() string {
+	return c.Method.Service.Pkg.Name
+}
+
+func (c *CodeSamples) ServiceName() string {
+	return c.Method.Service.Name
+}
+
+func (c *CodeSamples) MethodName() string {
+	return c.Method.Name
+}
+
+func (c *CodeSamples) ProtoFileName() string {
+	return c.Method.Service.FileName
+}
+
+func (c *CodeSamples) LoaderFiles() string {
+	if c.AppName() == "lnd" {
+		if c.ProtoFileName() == "lightning.proto" {
+			return "'lightning.proto'"
+		} else {
+			return fmt.Sprintf("['lightning.proto', '%s']", c.ProtoFileName())
```

Comment:
nit: no need for the `else` if there's a `return` in the `if`. (Golang really likes to have the happy path non-indented, so the use of `else` is discouraged wherever it helps readability).